src/components/ble/AppleNotificationCenterClient.cpp

```diff
@@ -0,0 +1,514 @@
+}
+
+void AppleNotificationCenterClient::RejectIncomingCall(uint32_t uuid) {
+  AncsNotification ancsNotif;
+  if (notifications.contains(uuid)) {
+    ancsNotif = notifications[uuid];
+    if (ancsNotif.category != static_cast<uint8_t>(Categories::IncomingCall)) {
+      return;
+    }
+  } else {
+    return;
+  }
+
+  uint8_t value[6];
+  value[0] = 0x02; // Command ID: Perform Notification Action
+  value[1] = (uint8_t) (uuid & 0xFF);
+  value[2] = (uint8_t) ((uuid >> 8) & 0xFF);
+  value[3] = (uint8_t) ((uuid >> 16) & 0xFF);
+  value[4] = (uint8_t) ((uuid >> 24) & 0xFF);
+  value[5] = 0x01; // Action ID: Negative Action
+
+  ble_gattc_write_flat(systemTask.nimble().connHandle(), controlPointHandle, value, sizeof(value), OnControlPointWriteCallback, this);
+}
+
+void AppleNotificationCenterClient::Reset() {
+  ancsStartHandle = 0;
+  ancsEndHandle = 0;
+  gattStartHandle = 0;
+  gattEndHandle = 0;
+  serviceChangedHandle = 0;
+  serviceChangedDescriptorHandle = 0;
+  notificationSourceHandle = 0;
+  notificationSourceDescriptorHandle = 0;
+  controlPointHandle = 0;
+  controlPointDescriptorHandle = 0;
+  dataSourceHandle = 0;
+  dataSourceDescriptorHandle = 0;
+  isGattDiscovered = false;
+  isGattCharacteristicDiscovered = false;
+  isGattDescriptorFound = false;
+  isDiscovered = false;
+  isCharacteristicDiscovered = false;
+  isDescriptorFound = false;
+  isControlCharacteristicDiscovered = false;
+  isControlDescriptorFound = false;
+  isDataCharacteristicDiscovered = false;
+  isDataDescriptorFound = false;
+
+  notifications.clear();
+}
+
+void AppleNotificationCenterClient::Discover(uint16_t connectionHandle, std::function<void(uint16_t)> onServiceDiscovered) {
+  NRF_LOG_INFO("[ANCS] Starting discovery");
+  this->onServiceDiscovered = onServiceDiscovered;
+  ble_gattc_disc_svc_by_uuid(connectionHandle, &ancsUuid.u, OnDiscoveryEventCallback, this);
+}
+
+void AppleNotificationCenterClient::DebugNotification(const char* msg) const {
+  NRF_LOG_INFO("[ANCS DEBUG] %s", msg);
+
+  NotificationManager::Notification notif;
+  std::strncpy(notif.message.data(), msg, notif.message.size() - 1);
+  notif.message[notif.message.size() - 1] = '\0'; // Ensure null-termination
+  notif.size = std::min(std::strlen(msg), notif.message.size());
+  notif.category = Pinetime::Controllers::NotificationManager::Categories::SimpleAlert;
+  notificationManager.Push(std::move(notif));
+
+  systemTask.PushMessage(Pinetime::System::Messages::OnNewNotification);
+}
+
+std::string AppleNotificationCenterClient::DecodeUtf8String(os_mbuf* om, uint16_t size, uint16_t offset) {
+  std::string decoded;
+  decoded.reserve(size);
+
+  auto isInFontDefinition = [](uint32_t codepoint) -> bool {
+    // Check if the codepoint falls into the specified font ranges or is explicitly listed
+    return (codepoint >= 0x20 && codepoint <= 0x7E) ||   // Printable ASCII
+           (codepoint >= 0x410 && codepoint <= 0x44F) || // Cyrillic
```

Comment:
How come there's a Cyrillic check? I thought this wasn't supported by any of the fonts. More generally do we need some generic way to check this is within the font ranges? How does the existing notification client handle this?